Tasks/google-play-promote/google-play-promote.ts

```diff
@@ -1,207 +1,73 @@
             }
-            let stats: fs.Stats = fs.statSync(serviceAccountKeyFile);
+            const stats: tl.FsStats = tl.stats(serviceAccountKeyFile);
             if (stats && stats.isFile()) {
                 key = require(serviceAccountKeyFile);
             } else {
                 console.error(tl.loc('InvalidAuthFile'));
                 throw new Error(tl.loc('InvalidAuthFilewithName', serviceAccountKeyFile));
             }
         } else if (authType === 'ServiceEndpoint') {
-            let serviceEndpoint: tl.EndpointAuthorization = tl.getEndpointAuthorization(tl.getInput('serviceEndpoint', true), true);
+            const serviceEndpoint: tl.EndpointAuthorization = tl.getEndpointAuthorization(tl.getInput('serviceEndpoint', true), true);
             if (!serviceEndpoint) {
                 throw new Error(tl.loc('EndpointNotFound'));
             }
             key.client_email = serviceEndpoint.parameters['username'];
             key.private_key = serviceEndpoint.parameters['password'].replace(/\\n/g, '\n');
         }
 
-        let packageName: string = tl.getPathInput('packageName', true);
-        let sourceTrack: string = tl.getInput('sourceTrack', true);
-        let destinationTrack: string = tl.getInput('destinationTrack', true);
-        let userFraction: number = Number(tl.getInput('userFraction', false)); // Used for staged rollouts
+        const packageName: string = tl.getPathInput('packageName', true);
+        const sourceTrack: string = tl.getInput('sourceTrack', true);
+        const destinationTrack: string = tl.getInput('destinationTrack', true);
+        const userFractionSupplied: boolean = tl.getBoolInput('rolloutToUserFraction');
+        const userFraction: number = Number(userFractionSupplied ? tl.getInput('userFraction', false) : 1.0);
 
         // Constants
-        let GOOGLE_PLAY_SCOPES: string[] = ['https://www.googleapis.com/auth/androidpublisher'];
-        let globalParams: GlobalParams = { auth: null, params: {} };
+        const globalParams: googleutil.GlobalParams = { auth: null, params: {} };
 
-        let jwtClient: any = new google.auth.JWT(key.client_email, null, key.private_key, GOOGLE_PLAY_SCOPES, null);
-        let edits: any = publisher.edits;
-
-        [edits, edits.tracks, jwtClient].forEach(Promise.promisifyAll);
+        const jwtClient = googleutil.getJWT(key);
+        const edits: any = googleutil.publisher.edits;
 
         globalParams.auth = jwtClient;
-        updateGlobalParams(globalParams, 'packageName', packageName);
+        googleutil.updateGlobalParams(globalParams, 'packageName', packageName);
 
         console.log(tl.loc('Authenticating'));
-        let currentEdit: any = jwtClient.authorizeAsync().then(function (res) {
-            return getNewEdit(edits, globalParams, packageName);
-        });
+        await jwtClient.authorize();
+        await googleutil.getNewEdit(edits, globalParams, packageName);
```

Comment:
This call does an additional `updateGlobalParams` in the new `getNewEdit` code path, is that okay?

Reply:
I'm not sure exactly what you mean.  The original `getNewEdit` in this class called `updateGlobalParams`.  I don't see an additional call.  (Unless I misread something.)

Reply:
(I am--in response to your other comment--pulling the line into a specific call to `updateGlobalParams`.)